### ISSUE TITLE
Habitat Hyperparam Confirmation

### ISSUE THREAD

**Issue by verityw** [facebookresearch/eai-vc#22]
It seems like some of the numbers reported for ObjectNav in the VC-1 paper are different from the ones listed in the config files. Namely, the paper says that the model was trained on ~400M steps for 25k updates on 512 parallel envs. However, the config files say that num_envs = 16 and num_updates = 12k.

Looking at the code, it seems like num updates should actually be more like the number of "epochs,"  wherein a batch of (num steps = 64, num envs) steps stored in the rollout buffer is used for training. This seems to be divided into num_mini_batch = 2 minibatches (each containing (num steps = 64, num envs / 2) steps).

Thus, the total number of gradient steps is number of "epochs"  * number of minibatches = 12k * 2 = 24k (which I guess is where the 25k number from the paper came from). If each update got a minibatch of (64, 512 / 2) steps, then the total number of steps trained for is 24k * 256 * 64 = 393M ~= 400M reported in the paper.

Also, I guess the 512 envs could come from DDP (eg if there is a world size of 32 and each spawned 16 envs, then the total would be 512). Haven't confirmed this in the code yet though.

Is all of that correct? Wanted to make sure my interpretation was alright.

**Comment by ykarmesh** [comment]
Hi @verityw,

Yes, your calculation is correct. We run our experiments on 4 nodes, each having 8 V100 GPUs (32 GPUs in total), which takes the total number of parallel environments to 512 (16 envs per GPU * 32 GPUs). 

Lets ignore the minibatching logic for now and calculate the total number of env steps per run.  Before each update, an agent collects 64 frames from per env. So there are 32768 (64 frames per env * 512 envs) frames collected in total. If there are a total of 12k updates, we get the 393M number (32768 frames per update * 12000 updates). You can also reach the same number if you including the minibatching step in the calculation.

In most prior papers I have seen, minibatching is usually ignored when reporting the number of updates. It has been ignored in the appendix for the ImageNav section, but it looks like we made a mistake in the ObjectNav section. Thanks for catching this and we will update the paper to say we do 12k updates.
